src/main/java/io/github/syst3ms/skriptparser/util/Date.java

```diff
@@ -0,0 +1,145 @@
+package io.github.syst3ms.skriptparser.util;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.text.SimpleDateFormat;
+import java.time.*;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * Represents a date.
+ * @author Njol
+ */
+public class Date implements Comparable<Date> {
+
+    // TODO make a config for this
+    public final static String DATE_FORMAT = "EEEE dd MMMM yyyy HH:mm:ss.SSS zzzXXX";
+    public final static String DATE_LOCALE = "US";
+
+    /**
+     * Timestamp. Should always be in computer time/UTC/GMT+0.
+     */
+    private long timestamp;
+
+    public Date() {
+        this(System.currentTimeMillis());
+    }
+
+    public Date(final long timestamp) {
```

Comment:
Njol adds `final`s everywhere, but that isn't the code style in this project. If they aren't used for (1) constants, (2) non-mutable fields, (3) non-overridable methods/non-extendible classes, they should be removed.

Njol did this thing a lot, and it should be taken care of when reusing his code.